### ISSUE TITLE
Defold games now crash when run via Proton (Could not find gl function 'glGenPrograms')

### ISSUE THREAD

**Issue by JustAPotota** [defold/defold#8767]
**Describe the bug (REQUIRED)**
Running a bundle of a Windows project on Linux via Proton makes the game open as a black window for a moment before closing. Native Linux bundles work as normal.

**To Reproduce (REQUIRED)**
1. Make a Windows bundle of a project
2. Add it to Steam on Linux
3. Set it to use Proton (any version) in the properties window
4. Run it

**Defold version (REQUIRED):**
Tested with 3 different projects built with 1.7.0 (a project of mine, the desktop template, and the [Chromacell demo](https://store.steampowered.com/app/2293310/Chromacell/)).

**Platforms (REQUIRED):**
- Fedora Linux 39
- Radeon RX 6700 XT GPU

**Logs (OPTIONAL):**
Logs taken from the desktop template.
- Defold log:  [log.txt](https://github.com/defold/defold/files/14898229/log.txt)

```
ERROR:GRAPHICS: Could not find gl function 'glGenPrograms'.
FATAL:ENGINE: Unable to create the graphics context.
```

- Proton log (via PROTON_LOG=1): [steam-11239744614107709440.log](https://github.com/defold/defold/files/14898232/steam-11239744614107709440.log)
- [_crash](https://github.com/defold/defold/files/14898284/_crash.txt)
- [_crash.dmp](https://github.com/defold/defold/files/14898286/_crash.dmp.txt)

**Additional context (OPTIONAL):**
A very old Windows bundle I have (Defold 1.2.171) still runs just fine on Proton.

This shouldn't be a big issue with how easy it is to bundle for other platforms with Defold, but 7 out of the 25 [Defold games on Steam](https://steamdb.info/instantsearch/?refinementList%5Btechnologies%5D%5B0%5D=Engine.Defold) are still Windows-only.

**Comment by Jhonnyg** [comment]
Could you try this build?
https://d.defold.com/archive/dev/300ed4080a65eda82423d60bd7a27bc21fdd55f0/dev/editor2/Defold-x86_64-linux.tar.gz

**Comment by JustAPotota** [comment]
Windows bundle still crashes, but now it can't find 'glBindProgram': [log.txt](https://github.com/defold/defold/files/14906024/log.txt)

Native Linux builds are broken now too (bundled or run in the editor), outputting:
```
INFO:DLIB: Log server started on port 37949
INFO:ENGINE: Target listening with name: goodpc - fe80::e687:d51f:cf33:5037 - Linux
INFO:ENGINE: Engine service started on port 39635
INFO:GRAPHICS: Installed graphics device 'ADAPTER_FAMILY_OPENGL'
INFO:ENGINE: Defold Engine 1.7.1 (300ed40)
```
...and then nothing happens. I can see the `engine_main` process is still running and using the CPU, but there's no window.

**Comment by Jhonnyg** [comment]
And this works with an olden defold still? Nothing has changed on your side regarding drivers and so forth?

**Comment by JustAPotota** [comment]
Yeah, running the desktop template from the editor consistently works on 1.7.0 and breaks on the build you sent.

**Comment by britzl** [comment]
Can you please verify that a Linux build works?

**Comment by JustAPotota** [comment]
Building the desktop template from editor on Linux:
- 1.7.0 stable (https://github.com/defold/defold/commit/bf4dc66ab5fbbafd4294d32c2797c08b630c0be5) works as it should
- 1.7.1 beta (https://github.com/defold/defold/commit/3964ada6b3fce0a219394b048b2466858bc19de9) is broken as mentioned above (game runs and takes up a whole CPU core, but no window, sound, or any errors)

Bundling the same project for Linux on the same 1.7.1 beta:
- Debug mode is still broken
- Release mode works perfectly (window and sound)

**Comment by AGulev** [comment]
@JustAPotota could you pls check it as well?

**Comment by JustAPotota** [comment]
Linux builds and bundles are back to normal now, but the original issue still remains.

**Comment by Jhonnyg** [comment]
So does 1.7.0 work as expected on the windows + proton setup?

**Comment by JustAPotota** [comment]
The original issue still persists on 1.7.0 and 1.8.0.

**Comment by JCash** [comment]
I have setup my steam deck accordingly, and I can also reproduce this issue.
I'll see if I can also use Visual Studio directly to make the turnaround times a bit better as well.

**Comment by AGulev** [comment]
@JustAPotota are you still able to test a build (I mean windows build on Linux using proton)?
If so, it would be awesome if you can try this one 

https://d.defold.com/archive/dev/5800036c295b13f18e810db387edfdf9e73e8395/dev/editor2/Defold-x86_64-win32.zip

https://d.defold.com/archive/5800036c295b13f18e810db387edfdf9e73e8395/dev/editor2/Defold-x86_64-linux.zip

**Comment by JustAPotota** [comment]
Seems to work! I tested both of those editors with a random project, and the resulting Windows builds run just fine under all the recent versions of Proton I tried.

**Comment by AGulev** [comment]
Thank you for verification!
I have prepared PR for the issue
It will be closed after merge
